depth.py

```diff
@@ -1,28 +1,90 @@
+import argparse
 import sys
-import parse_gfa
+import odgi
+''' 
+The documentation for the odgi module can be found at 
+https://odgi.readthedocs.io/en/latest/rst/binding/glossary.html 
+'''
 
-def depth(filename):
+def get_depth_table(graph, subset_paths=None):
+    ''' 
+    Input: an odgi.graph object
+    Output: the node depth table, a dictionary that maps from a node's id to it's (depth, uniq_depth),
+        where depth is the total number of times each path in subset_paths crosses the node,
+        and uniq_depth is the number of paths in subset_paths which cross the node
+    Node: if subset_paths is empty, consider all paths when computing node depth
     '''
-    Calculates the depth of each node by counting how many times a node appears
-    in graph
-    '''
-    g = parse_gfa.parse_file_path_only(filename)
-    depth_map = {}
-
-    for path in g.paths:
-        for segment in path.segment_names:
-            name = int(segment.name)
-            if name in depth_map:
-                depth_map[name] += 1
-            else:
-                depth_map[name] = 1
-    sorted_depth_items = sorted(depth_map.items())
-
-    with open('python_output.txt', 'w') as f: 
-        for pair in sorted_depth_items:
-            f.write(f'{pair[0]} {pair[1]}\n')
+
+    ndt = dict() # node depth table map from node.id -> (node.depth, node.uniq_depth)
+
+    # Compute the set of paths we want to consider
+    if subset_paths is None: # By default, we want to consider all paths
+        paths_to_consider = set()
+        # Adds the name of each path in graph to paths_to_consider
+        graph.for_each_path_handle(lambda h: paths_to_consider.add(graph.get_path_name(h)))
+    else: # Consider only paths specified, if a subet of paths is specified
+        paths_to_consider = set(subset_paths)
+
+    # Compute the node depth and unique depth
+    def get_node_depth(handle):
+        '''
+        Input: [handle] is an odgi.handle object which representing a node
+        Inserts node.depth and node.uniq into ndt for the node associated with
+            [handle]
+        '''
+
+        # Note: a node can have multiple handles, but only one id
```

Comment:
Looking at the function [`get_handle`](https://github.com/pangenome/odgi/blob/e2de6cbca0169b0720dca0c668743399305e92bd/src/odgi.cpp#L29-L31), which returns a node handle based on the node's id, I believe that a node handle encodes (1) a node id and (2) a boolean indicating whether to traverse the node backwards or forwards. Since the node id is unique, this would mean there can be at most two node handles per node.

Reply:
That would explain it! Thanks!